External/CSG/Classes/CSG_Model.cs

```diff
@@ -0,0 +1,139 @@
+using System;
+using UnityEngine;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace UnityEngine.ProBuilder.Csg
+{
+    /// <summary>
+    /// Representation of a mesh in CSG terms. Contains methods for translating to and from UnityEngine.Mesh.
+    /// </summary>
+    sealed class CSG_Model
+    {
+        internal List<CSG_Vertex> vertices;
+        List<Material> m_Materials;
+        List<List<int>> m_Indices;
+
+        public List<Material> materials
```

Comment:
This could potentially returns  null if we only create a new CSG_Model from its constructor.
Ctor instantiates `m_Indices` and `vertices` but not `m_Materials`, that seems inconsistent.